crates/uv/tests/it/init.rs

```diff
@@ -929,6 +929,66 @@ fn init_script_file_conflicts() -> Result<()> {
     Ok(())
 }
 
+// Init script should not trash an existing shebang.
+#[test]
+fn init_script_shebang() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let child = context.temp_dir.child("foo");
+    child.create_dir_all()?;
+
+    let contents = "#! /usr/bin/env python3\nprint(\"Hello, world!\")";
+    fs_err::write(child.join("script.py"), contents)?;
+    uv_snapshot!(context.filters(), context.init().current_dir(&child).arg("--script").arg("script.py"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    warning: If you execute script.py directly, it might ignore its inline metadata.
+    warning: Consider replacing its shebang with: #!/usr/bin/env -S uv run --script
+    Initialized script at `script.py`
+    ");
+    let resulting_script = fs_err::read_to_string(child.join("script.py"))?;
+    assert_snapshot!(resulting_script, @r#"
+    #! /usr/bin/env python3
+
+    # /// script
+    # requires-python = ">=3.12"
+    # dependencies = []
+    # ///
+
+    print("Hello, world!")
+    "#
+    );
+
+    // If the shebang already contains `uv`, the result is the same, but we suppress the warning.
+    let contents = "#!/usr/bin/env -S uv run --script\nprint(\"Hello, world!\")";
+    fs_err::write(child.join("script.py"), contents)?;
+    uv_snapshot!(context.filters(), context.init().current_dir(&child).arg("--script").arg("script.py"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Initialized script at `script.py`
+    ");
+    let resulting_script = fs_err::read_to_string(child.join("script.py"))?;
+    assert_snapshot!(resulting_script, @r#"
+    #! /usr/bin/env python3
+
```

Comment:
I'm not sure I love the newline without a leading comment character, I might include a `#`?